File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalMonthlyDayScheduleBridge.java
package com.krystianwsul.checkme.domainmodel.local;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.krystianwsul.checkme.domainmodel.MonthlyDayScheduleBridge;
import com.krystianwsul.checkme.persistencemodel.MonthlyDayScheduleRecord;
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord;
import com.krystianwsul.checkme.utils.CustomTimeKey;

class LocalMonthlyDayScheduleBridge extends LocalScheduleBridge implements MonthlyDayScheduleBridge {
    @NonNull
    private final MonthlyDayScheduleRecord mMonthlyDayScheduleRecord;

    LocalMonthlyDayScheduleBridge(@NonNull ScheduleRecord scheduleRecord, @NonNull MonthlyDayScheduleRecord monthlyDayScheduleRecord) {
        super(scheduleRecord);

        mMonthlyDayScheduleRecord = monthlyDayScheduleRecord;
    }

    @Override
    public int getDayOfMonth() {
        return mMonthlyDayScheduleRecord.getDayOfMonth();
    }

    @Override
    public boolean getBeginningOfMonth() {
        return mMonthlyDayScheduleRecord.getBeginningOfMonth();
    }

    @Nullable
    @Override
    public CustomTimeKey getCustomTimeKey() {
        if (mMonthlyDayScheduleRecord.getCustomTimeId() != null)
            return new CustomTimeKey(mMonthlyDayScheduleRecord.getCustomTimeId());
        else
            return null;
    }

    @Nullable
    @Override
    public Integer getHour() {
        return mMonthlyDayScheduleRecord.getHour();
    }

    @Nullable
    @Override
    public Integer getMinute() {
        return mMonthlyDayScheduleRecord.getMinute();
    }

    @Override
    public void delete() {
        getScheduleRecord().delete();
        mMonthlyDayScheduleRecord.delete();
    }
}

File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalMonthlyDayScheduleBridge.kt
package com.krystianwsul.checkme.domainmodel.local

import com.krystianwsul.checkme.domainmodel.MonthlyDayScheduleBridge
import com.krystianwsul.checkme.persistencemodel.MonthlyDayScheduleRecord
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord
import com.krystianwsul.checkme.utils.CustomTimeKey

class LocalMonthlyDayScheduleBridge(scheduleRecord: ScheduleRecord, private val monthlyDayScheduleRecord: MonthlyDayScheduleRecord) : LocalScheduleBridge(scheduleRecord), MonthlyDayScheduleBridge {

    override val dayOfMonth get() = monthlyDayScheduleRecord.dayOfMonth

    override val beginningOfMonth get() = monthlyDayScheduleRecord.beginningOfMonth

    override val customTimeKey get() = monthlyDayScheduleRecord.customTimeId?.let { CustomTimeKey.LocalCustomTimeKey(it) }

    override val hour get() = monthlyDayScheduleRecord.hour

    override val minute get() = monthlyDayScheduleRecord.minute

    override fun delete() {
        scheduleRecord.delete()
        monthlyDayScheduleRecord.delete()
    }
}
